### PR TITLE
Cleanup any stale overlay bridge with overlapping subnet in hostMode

### PR DESCRIPTION
**Pull request by mavenugo**
Signed-off-by: Madhu Venugopal <madhu@docker.com>

### CODE REVIEW

**Comment by mavenugo** [comment]
ping @mrjana

**Comment by mrjana** [comment]
LGTM

File: drivers/overlay/ov_network.go [hunk 1]
```diff
@@ -388,7 +392,9 @@ func (n *network) initSubnetSandbox(s *subnet) error {
 
 	if hostMode {
 		// Try to delete stale bridge interface if it exists
-		deleteInterface(brName)
+		if err := deleteInterface(brName); err != nil {
+			deleteInterfaceBySubnet(n.getBridgeNamePrefix(s), s)
```
Comment (aboch) [inline drivers/overlay/ov_network.go]: It would be good to log a warn if the cleanup did not succeeded

File: drivers/overlay/ov_utils.go [hunk 2]
```diff
@@ -69,6 +71,37 @@ func createVxlan(name string, vni uint32) error {
 	return nil
 }
 
+func deleteInterfaceBySubnet(brPrefix string, s *subnet) error {
+	defer osl.InitOSContext()()
+
+	nlh := ns.NlHandle()
+	links, err := nlh.LinkList()
+	if err != nil {
+		return fmt.Errorf("failed to list interfaces while deleting bridge interface by subnet: %v", err)
```
Comment (aboch) [inline drivers/overlay/ov_utils.go]: or just make this a log. The cleanup is a best effort. then the function do not need to return any error.

**Comment by aboch** [comment]
LGTM besides a minor comment

**Pull request merged**